packages/discovery-react-components/src/components/StructuredQuery/utils/__tests__/getFieldNames.test.ts

```diff
@@ -0,0 +1,49 @@
+import { getFieldNames } from '../getFieldNames';
+
+function matchingArrays(array1: string[], array2: string[]): boolean {
```

Comment:
nit - since these are arrays of primitives, you could just compare the strings -> array1.toString() === array2.toString()

Reply:
another good call, thanks!